walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,214 @@
+    Collection<String> mandatory = request.getAttendees();
+    long duration = request.getDuration();
+    List<TimeRange> possibleTimes = new ArrayList<>();
+
+    int prev_end = 0;
+      for (Event e : events_list){
+        // Determine if the attendees for the meeting are busy at other events
+        Collection<String> intersection = new HashSet<>(mandatory);
+        intersection.retainAll(e.getAttendees());
+        TimeRange range = e.getWhen();
+        if (intersection.size() > 0) {
+          TimeRange possibleRange = TimeRange.fromStartEnd(prev_end, range.start(), false);
+          if (duration <= possibleRange.duration()) {
+            possibleTimes.add(possibleRange);
+          }
+          if (prev_end < range.end()) {
+            prev_end = range.end();
+          }
+        }
+      }
+      if (prev_end != TimeRange.END_OF_DAY + 1) {
+        possibleTimes.add(TimeRange.fromStartEnd(prev_end, TimeRange.END_OF_DAY, true));
+      }
+      return possibleTimes;
+  }
+
+  /**
+   * Finds a specific attendee's events for the day.
+   * @param events_list sorted list of events in ascending start time order
+   * @param attendee specific attendee to search for
+   * @return collection of events that attendee has on the schedule
+   */
+  private Collection<Event> eventFinderWithAttendee(Collection<Event> events_list, String attendee) {
+    List<Event> eventsByAttendee = new ArrayList<Event>(); 
+
+    for (Event e : events_list){
+      if (e.getAttendees().contains(attendee)){
+        eventsByAttendee.add(e);
+      }
+    }
+    return eventsByAttendee;
+  }
+
+  /**
+   * Split large range into smaller pieces without event overlapping
+   * Range:  |-------------|
+   * Event:       |---|
+   * Return: |----|   |----|
+   * @param sorted_events sorted list of events in ascending start time order
+   * @param range large range to split into smaller ranges
+   * @param duration how long the request is for
+   * @return collection of TimeRanges that was part of the large range without event overlaps
+   */
+  private Collection<TimeRange> rangeSplitByNestedEvents(Collection<Event> sorted_events, TimeRange range, long duration) {
+    List<TimeRange> splitEvents = new ArrayList<TimeRange>(); 
+    int prevEndTime = range.start();
+
+    for (Event e : sorted_events) {
+      if (range.overlaps(e.getWhen())) {
+        if (range.start() <= e.getWhen().start()) {
+          TimeRange begin = TimeRange.fromStartEnd(prevEndTime, e.getWhen().start(), false);
+          if (begin.duration() >= duration){
+            splitEvents.add(begin);
+          }
+        }
+        prevEndTime = e.getWhen().end();
+
+      }
+    }
+    if (prevEndTime < range.end()) {
+      splitEvents.add(TimeRange.fromStartEnd(prevEndTime, range.end(), false));
+    }
+    return splitEvents;
+  }
+
+  /**
+   * Counts the number of optional attendees can not make it to the meeting at the time range
```

Comment:
Comment describes what's happening in the code but doesn't clarify why it's comparing these two numbers? Hard to understand purpose of function